src/Formik.tsx

```diff
@@ -561,62 +573,61 @@ export function useFormik<Values = object>({
 
         if (field) {
           // Set form fields by name
-          dispatch({ type: 'SET_FIELD_VALUE', payload: { field, value: val } });
+          setFieldValue(field, val);
         }
       }
     },
-    []
+    [setFieldValue]
   );
 
-  const setTouched = React.useCallback((touched: FormikTouched<Values>) => {
-    dispatch({ type: 'SET_TOUCHED', payload: touched });
-  }, []);
-
-  const setErrors = React.useCallback((errors: FormikErrors<Values>) => {
-    dispatch({ type: 'SET_ERRORS', payload: errors });
-  }, []);
-
-  const setValues = React.useCallback((values: Values) => {
-    dispatch({ type: 'SET_VALUES', payload: values });
-  }, []);
-
-  const setFieldError = React.useCallback(
-    (field: string, value: string | undefined) => {
+  const setFieldTouched = React.useCallback(
+    (
+      field: string,
+      touched: boolean = true,
+      shouldValidate: boolean = true
+    ) => {
       dispatch({
-        type: 'SET_FIELD_ERROR',
-        payload: { field, value },
+        type: 'SET_FIELD_TOUCHED',
+        payload: {
+          field,
+          value: touched,
+        },
       });
+      return validateOnBlur && shouldValidate
+        ? validateFormWithLowPriority(state.values)
+        : Promise.resolve();
     },
-    []
+    [validateFormWithLowPriority, state.values, validateOnBlur]
   );
 
-  const setFieldValue = React.useCallback((
-    field: string,
-    value: any
-    // shouldValidate: boolean = true
-  ) => {
-    dispatch({
-      type: 'SET_FIELD_VALUE',
-      payload: {
-        field,
-        value,
-      },
-    });
-  }, []);
+  const handleBlur = React.useCallback(
+    (eventOrString: any): void | ((e: any) => void) => {
+      if (isString(eventOrString)) {
+        return event => executeBlur(event, eventOrString);
+      } else {
+        executeBlur(eventOrString);
+      }
 
-  const setFieldTouched = React.useCallback((
-    field: string,
-    touched: boolean = true
-    // shouldValidate: boolean = true
-  ) => {
-    dispatch({
-      type: 'SET_FIELD_TOUCHED',
-      payload: {
-        field,
-        value: touched,
-      },
-    });
-  }, []);
+      function executeBlur(e: any, path?: string) {
+        if (e.persist) {
+          e.persist();
+        }
+        const { name, id, outerHTML } = e.target;
+        const field = path ? path : name ? name : id;
```

Comment:
`path || name || id`?

Reply:
Use the suggestions feature next time so it's easier for maintainers.
```suggestion
        const field = path || name || id;
```

Reply:
Personally I would limit this to required `name` prop and that's it... @jaredpalmer What do you think? Enforce `name` prop on fields for v2?

Reply:
I forgot when it happened exactly, but we used to support `id` only. I think things got weird with FieldArray IIRC. I just want us to be consistent.

Reply:
@Andreyco lets move this to a new issue

Reply:
See #1558